python/tests/expressions/test_visitors.py

```diff
@@ -1103,363 +1091,296 @@ def test_or(schema: Schema, manifest: ManifestFile):
-    ).eval(manifest), "Should read: id equal to lower bound (30 == 30)"
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("id"), (literal(INT_MIN_VALUE - 1), literal(INT_MIN_VALUE)))).eval(
+        manifest
+    ), "Should read: id equal to lower bound (30 == 30)"
 
-    assert _ManifestEvalVisitor(
-        schema, NotIn(Reference[int]("id"), (LongLiteral(INT_MAX_VALUE - 4), LongLiteral(INT_MAX_VALUE - 3)))
-    ).eval(manifest), "Should read: id between lower and upper bounds (30 < 75 < 79, 30 < 76 < 79)"
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("id"), (literal(INT_MAX_VALUE - 4), literal(INT_MAX_VALUE - 3)))).eval(
+        manifest
+    ), "Should read: id between lower and upper bounds (30 < 75 < 79, 30 < 76 < 79)"
 
-    assert _ManifestEvalVisitor(
-        schema, NotIn(Reference[int]("id"), (LongLiteral(INT_MAX_VALUE), LongLiteral(INT_MAX_VALUE + 1)))
-    ).eval(manifest), "Should read: id equal to upper bound (79 == 79)"
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("id"), (literal(INT_MAX_VALUE), literal(INT_MAX_VALUE + 1)))).eval(
+        manifest
+    ), "Should read: id equal to upper bound (79 == 79)"
 
-    assert _ManifestEvalVisitor(
-        schema, NotIn(Reference[int]("id"), (LongLiteral(INT_MAX_VALUE + 1), LongLiteral(INT_MAX_VALUE + 2)))
-    ).eval(manifest), "Should read: id above upper bound (80 > 79, 81 > 79)"
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("id"), (literal(INT_MAX_VALUE + 1), literal(INT_MAX_VALUE + 2)))).eval(
+        manifest
+    ), "Should read: id above upper bound (80 > 79, 81 > 79)"
 
-    assert _ManifestEvalVisitor(
-        schema, NotIn(Reference[int]("id"), (LongLiteral(INT_MAX_VALUE + 6), LongLiteral(INT_MAX_VALUE + 7)))
-    ).eval(manifest), "Should read: id above upper bound (85 > 79, 86 > 79)"
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("id"), (literal(INT_MAX_VALUE + 6), literal(INT_MAX_VALUE + 7)))).eval(
+        manifest
+    ), "Should read: id above upper bound (85 > 79, 86 > 79)"
 
-    assert _ManifestEvalVisitor(
-        schema, NotIn(Reference[str]("all_nulls_missing_nan"), (StringLiteral("abc"), StringLiteral("def")))
-    ).eval(manifest), "Should read: notIn on no nulls column"
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("all_nulls_missing_nan"), (literal("abc"), literal("def")))).eval(
+        manifest
+    ), "Should read: notIn on no nulls column"
 
-    assert _ManifestEvalVisitor(schema, NotIn(Reference[str]("some_nulls"), (StringLiteral("abc"), StringLiteral("def")))).eval(
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("some_nulls"), (literal("abc"), literal("def")))).eval(
         manifest
     ), "Should read: in on some nulls column"
 
-    assert _ManifestEvalVisitor(schema, NotIn(Reference[str]("no_nulls"), (StringLiteral("abc"), StringLiteral("def")))).eval(
+    assert _ManifestEvalVisitor(schema, NotIn(Reference("no_nulls"), (literal("abc"), literal("def")))).eval(
```

Comment:
As long as we're modifying these lines, why not remove `literal` and let the constructor handle it? (Not a blocker, just curious)

Reply:
Ah, that would also be a possibility. I just wanted to get the explicit types out of the way